EosPayload/lib/driver_base.py

```diff
@@ -1,116 +1,267 @@
+        """
         self.spin()
 
-    def device_command(self, logger: logging.Logger) -> None:
+    def device_command(self, logger: logging.Logger, shared_queue: Queue) -> None:
+        """ [OPTIONAL] Main function for Driver Command Thread.
+        Used to write to device.
+        Method should not return, which would terminate the thread.  Use self.spin() to keep alive.
+
+        :param logger: can be used to log info / error messages to disk and console
+        :param shared_queue: a queue that can be used to communicate between threads
+        """
         self.spin()
 
+    #
+    # UTILITY METHODS
+    #
+
     @staticmethod
     def spin() -> None:
+        """ Loops forever. """
         while True:
             time.sleep(1)
 
-    @staticmethod
-    @abstractmethod
-    def get_device_id() -> str:
-        pass
+    def is_healthy(self) -> bool:
+        """ Reports if the driver is operating properly
 
-    @staticmethod
-    def enabled() -> bool:
-        return True
+        :return: True if all threads are alive, False otherwise
+        """
 
-    def is_healthy(self) -> bool:
         return self.__read_thread.is_alive() and self.__command_thread.is_alive()
 
-    def send_heartbeat(self, mqtt: Client) -> bool:
+    def __send_heartbeat(self, mqtt: Client) -> bool:
+        """ Dispatches a heartbeat message to MQTT.
+        Should not be overriden or invoked by subclasses.
+
+        :param mqtt: a connected MQTT client
+        :return: True if heartbeat successfully sent, False otherwise
+        """
         succeeded = False
         if mqtt:
-            succeeded = mqtt.send(Topic.HEALTH_HEARTBEAT, self.get_device_id())
+            status = ','.join([
+                str(int(self.get_device_id())),
+                datetime.now().isoformat(),
+                str(int(self.is_healthy())),
+                str(int(self.__read_thread.is_alive())),
+                str(int(self.__command_thread.is_alive()))
+            ])
+            succeeded = mqtt.send(Topic.HEALTH_HEARTBEAT, status)
         if succeeded:
-            self.logger.info("heartbeat sent")
+            self.__logger.info("heartbeat sent")
         else:
-            self.logger.warning("heartbeat failed to send")
+            self.__logger.warning("heartbeat failed to send")
         return succeeded
 
-    def data_log(self, data: list[str]) -> None:
+    def data_log(self, data: list[str]) -> bool:
+        """ Logs row of data to a CSV file
+        This function is not thread safe -- invoke from at most 1 thread.
+
+        :param data: an array of strings
+        :return: True on success, False otherwise
+        """
         timestamp = datetime.now().isoformat()
         data_str = ','.join([timestamp, *data]) + "\n"
         self.__data_file.write(data_str)
         self.__data_file.flush()
+        return True
+
+    def data_transmit(self, mqtt: Client, data: list[str], telemetry: bool = False) -> bool:
```

Comment:
I've decided against this.  The implementation of data_emit as you describe would essentially be an alias to mqtt.send, since packetization would be delegated to the caller.  It is better to retain the existing data_transmit as it covers the general case of sending basic CSV data.  More complicated use cases can build the packet and call mqtt.send directly instead